### PR TITLE
Automate configuring optimizer and scheduler from config in scripts

### PR DESCRIPTION
**Pull request by dhpitt**
Refactors some old repeated logic out of the training recipe scripts and puts it in `neuralop/training/torch_setup.py`

### CODE REVIEW

File: neuralop/training/torch_setup.py [hunk 1]
```diff
@@ -91,6 +91,47 @@ def setup(config):
 
     return device, is_logger
 
+def config_opt_and_scheduler(config, model):
```
Comment (JeanKossaifi) [inline neuralop/training/torch_setup.py:94]: This looks too opaque to me : can we just have get_optimizer and get_scheduler methods that explicitly list their options?

File: neuralop/training/torch_setup.py [hunk 2]
```diff
@@ -91,6 +91,47 @@ def setup(config):
...
+            optimizer, T_max=config.opt.scheduler_T_max
+        )
+    elif config.opt.scheduler == "StepLR":
+        scheduler = torch.optim.lr_scheduler.StepLR(
+            optimizer, step_size=config.opt.step_size, gamma=config.opt.gamma
+        )
+    else:
+        raise ValueError(f"Got scheduler={config.opt.scheduler}")
+
+    return optimizer, scheduler
```
Comment (JeanKossaifi) [inline neuralop/training/torch_setup.py:133]: All this can be reused though

**Review by JeanKossaifi — COMMENTED** [review]
Minor suggestion to make code clearer, looks good otherwise.

File: neuralop/training/torch_setup.py (hunk 1 above)
Comment (dhpitt) [reply neuralop/training/torch_setup.py:94]: Sure, that sounds simpler

**Comment by dhpitt** [comment]
One question @JeanKossaifi - should these functions be in `torch_setup` or should they be in `neuralop.utils`?

**Comment by dhpitt** [comment]
Closing as the code is simple enough as-is.
